app/services/teachers/school_transfers/transfer.rb

```diff
@@ -0,0 +1,30 @@
+module Teachers::SchoolTransfers
+  class InvalidTransferError < StandardError; end
+
+  Transfer = Data.define(
+    :leaving_training_period,
+    :leaving_school,
+    :joining_training_period,
+    :joining_school
+  ) do
+    def type
+      return :unknown unless joining_training_period
+      return :new_provider if leaving_training_period.lead_provider !=
+        joining_training_period.lead_provider
+      return :new_school if leaving_school != joining_school
+
+      raise InvalidTransferError, "Unexpected transfer"
+    end
+
+    def status
+      if leaving_training_period.finished_on.future? ||
+          joining_training_period&.started_on&.future?
+        :incomplete
+      else
+        :complete
+      end
+    end
+
+    def teacher_type = leaving_training_period.for_ect? ? :ect : :mentor
```

Comment:
Maybe instead of `teacher_type` we should just delegate `for_ect?` and `for_mentor?` to the leaving training period?